File: com/pdfjet/Title.java
/*
 * Title.java
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

/**
 * Please see Example_48
 */
public class Title implements Drawable {
    TextLine prefix;
    TextLine textLine;

    private float offset = 0f;

    /**
     * Creates a title.
     *
     * @param font the font.
     * @param title the title text.
     * @param x the x coordinate.
     * @param y the y coordinate.
     */
    public Title(Font font, String title, float x, float y) {
        this.prefix = new TextLine(font);
        this.prefix.setLocation(x, y);
        this.textLine = new TextLine(font, title);
        this.textLine.setLocation(x, y);
    }

    /**
     * Sets the prefix text.
     *
     * @param text the prefix text.
     * @return this Title object.
     */
    public Title setPrefix(String text) {
        prefix.setText(text);
        return this;
    }

    /**
     * Sets the distance from the start of the prefix to the start of the title
     * text, to make room for the prefix.
     *
     * @param offset the offset.
     * @return this Title object.
     */
    public Title setOffset(float offset) {
        this.offset = offset;
        textLine.setLocation(prefix.x + offset, prefix.y);
        return this;
    }

    /**
     * Returns the prefix drawn before the title text, such as a section number.
     *
     * @return the prefix text line.
     */
    public TextLine getPrefix() {
        return prefix;
    }

    /**
     * Returns the title text.
     *
     * @return the title text line.
     */
    public TextLine getTextLine() {
        return textLine;
    }

    /**
     * Sets the location of the prefix; the title text keeps its offset from it.
     *
     * @param x the x coordinate.
     * @param y the y coordinate.
     * @return this Title object.
     */
    public Title setLocation(float x, float y) {
        prefix.setLocation(x, y);
        textLine.setLocation(x + offset, y);
        return this;
    }

    public float[] drawOn(Page page) throws Exception {
        if (!prefix.equals("")) {
            prefix.drawOn(page);
        }
        return textLine.drawOn(page);
    }
}

File: src/title.go
// title.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

package pdfjet

// Title is used to create title objects that have prefix and text.
// Please see Example_48
type Title struct {
	prefix   *TextLine
	textLine *TextLine
	offset   float32
}

// NewTitle is the constructor.
func NewTitle(font *Font, text string, x, y float32) *Title {
	title := new(Title)
	title.prefix = NewTextLine(font, "")
	title.textLine = NewTextLine(font, text)
	title.prefix.SetLocation(x, y)
	title.textLine.SetLocation(x, y)
	return title
}

// SetPrefix sets the prefix text.
func (title *Title) SetPrefix(text string) *Title {
	title.prefix.SetText(text)
	return title
}

// GetPrefix returns the prefix of the title.
func (title *Title) GetPrefix() *TextLine {
	return title.prefix
}

// GetTextLine returns the text line of the title.
func (title *Title) GetTextLine() *TextLine {
	return title.textLine
}

// SetOffset sets the distance from the start of the prefix to the start of the
// title text, to make room for the prefix.
func (title *Title) SetOffset(offset float32) *Title {
	title.offset = offset
	title.textLine.SetLocation(title.prefix.x+offset, title.prefix.y)
	return title
}

// SetLocation sets the location of the prefix; the title text keeps its offset
// from it.
//   - x: the x coordinate.
//   - y: the y coordinate.
//
// Returns this Title.
func (title *Title) SetLocation(x, y float32) Drawable {
	title.prefix.SetLocation(x, y)
	title.textLine.SetLocation(x+title.offset, y)
	return title
}

// DrawOn draws the title.
func (title *Title) DrawOn(page *Page) [2]float32 {
	if title.prefix.text != "" {
		title.prefix.DrawOn(page)
	}
	return title.textLine.DrawOn(page)
}
